Automattic/pocket-casts-ios
# Transcripts: add player transition to Transcripts

Part of #1848

This adds a transition in the player, allowing transcripts to be displayed while the play controls are still present.

Notice that we haven't defined yet the point of entry to the feature.

https://github.com/Automattic/pocket-casts-ios/assets/7040243/fe7ea98a-98f3-4978-8207-a661a6f39d53

Important: the transition here is not strictly the same as the [one here](https://pocketcastsp2.wordpress.com/2024/04/29/transcripts-i1/#a-icon-toggle). There are still tweaks to be done, but I've spent quite some time on this task for now. I believe the current state allows the work to proceed and then we can refine transitions once the bulk part of the work is done.

Important pt 2: During scoping, we discussed migration of the player to SwiftUI but that will be a complex task and a project on its own.

## To test

Change `playPauseTapped` in `NowPlayingPlayerItemViewController.swift` to:

```swift
    @IBAction func playPauseTapped(_ sender: Any) {
        displayTranscript.toggle()
    }
```

1. Run the app
2. If the miniplayer is visible, tap it. If not, play any episode
3. Tap the play/pause button on the full player
4. ✅ Notice how the player controls and the progress bar goes to the bottom and the tabs disappear
5. Test on small iPhone, big iPhone and iPad

## Checklist
